src/main/java/seedu/address/model/calendarevent/DateTime.java

```diff
@@ -1,108 +1,123 @@
+        if (!DateTime.isValidDateTime(year, month, day, hour, minute)) {
+            throw new DateTimeException(DateTime.MESSAGE_DATETIME_CONSTRAINTS);
         }
-    }
-
-    public int getYear() {
-        return date.getYear();
-    }
 
-    public int getMonth() {
-        return date.getMonthValue();
+        this.localDateTime = LocalDateTime.of(year, month, day, hour, minute);
     }
 
-    public int getDay() {
-        return date.getDayOfMonth();
-    }
-
-    public int getHour() {
-        return date.getHour();
-    }
-
-    public int getMinute() {
-        return date.getMinute();
-    }
-
-    public void setYear(int year) throws DateTimeException {
-        date = date.withYear(year);
-    }
+    /**
+     * Constructs a {@code DateTime} from date time input string
+     * Wrapper class for LocalDateTime
+     *
+     * @param dateTimeInput     valid date time input string
+     */
+    public DateTime(String dateTimeInput) {
+        String[] dateTimeInputNumbers = dateTimeInput.split("[- :]");
 
-    public void setMonth(int month) throws DateTimeException {
-        date = date.withMonth(month);
-    }
+        int year = Integer.parseInt(dateTimeInputNumbers[0]);
+        int month = Integer.parseInt(dateTimeInputNumbers[1]);
+        int day = Integer.parseInt(dateTimeInputNumbers[2]);
+        int hour = Integer.parseInt(dateTimeInputNumbers[3]);
+        int minute = Integer.parseInt(dateTimeInputNumbers[4]);
 
-    public void setDay(int day) throws DateTimeException {
-        date = date.withDayOfMonth(day);
-    }
+        if (!DateTime.isValidDateTime(year, month, day, hour, minute)) {
+            throw new DateTimeException(DateTime.MESSAGE_DATETIME_CONSTRAINTS);
+        }
 
-    public void setHour(int hour) throws DateTimeException {
-        date = date.withHour(hour);
+        this.localDateTime = LocalDateTime.of(year, month, day, hour, minute);
     }
 
-    public void setMinute(int minute) {
-        date = date.withMinute(minute);
-    }
+    public String toInputFormat() { return localDateTime.format(DateTimeFormatter.ofPattern("yyyy-MM-dd HH:mm")); }
 
     @Override
     public String toString() {
-        return localDateTime.toString();
+        return localDateTime.format(DateTimeFormatter.ofPattern("dd MMM yyyy hh:mm a"));
+    }
+
+    /**
+     * Returns true if a given string is a valid date time input string.
+     */
+    public static boolean isValidDateTimeInput(String test) {
+        return test.matches(DATETIMEINPUT_VALIDATION_REGEX);
     }
 
     /**
-     * Returns if a given datetime is a valid datetime.
+     * Returns if a given year, month, day, hour and minute correspond to a valid datetime.
      */
     public static boolean isValidDateTime(int year, int month, int day, int hour, int minute) {
-        try {
-            LocalDateTime dateTime = LocalDateTime.of(year, month, day, hour, minute);
-            DateTime dateTime1 = new DateTime(dateTime);
```

Comment:
in fact, this one can simply check the validation of number.

Reply:
Changed it so that it's using LocalDateTime to validate.